src/annotator/anchoring/types.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * This module exports a set of classes for converting between DOM `Range`
+ * objects and different types of selectors. It is mostly a thin wrapper around a
+ * set of anchoring libraries. It serves two main purposes:
+ *
+ *  1. Providing a consistent interface across different types of anchors.
+ *  2. Insulating the rest of the code from API changes in the underlying anchoring
+ *     libraries.
+ */
+import {
+  fromRange as posFromRange,
+  toRange as posToRange,
+} from 'dom-anchor-text-position';
+import {
+  fromRange as quoteFromRange,
+  toRange as quoteToRange,
+  toTextPosition,
+} from 'dom-anchor-text-quote';
+
+import { SerializedRange, sniff } from './range';
+
+/**
+ * @typedef {import("./range").BrowserRange} BrowserRange}
+ * @typedef {import("./range").NormalizedRange} NormalizedRange}
+ * @typedef {BrowserRange|NormalizedRange|SerializedRange} AnyRangeType
```

Comment:
Should `AnyRangeType` also accept an ordinary DOM `Range` as you use in several of your tests or does it have to be one of these three classes?

Reply:
Actually, after converting html over to js. I think it needs to also needs to support Range. I will double check.